pkg/synthetictests/duplicated_events.go

```diff
@@ -389,6 +390,75 @@ func getTimesAnEventHappened(message string) (string, int) {
 	return matches[0][1], int(times)
 }
 
+func getInstallCompletionTime(kubeClientConfig *rest.Config) *metav1.Time {
+	configClient, err := configclient.NewForConfig(kubeClientConfig)
+	if err == nil {
+		clusterVersion, err := configClient.ConfigV1().ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
+		if err == nil && len(clusterVersion.Status.History) != 0 {
+			return clusterVersion.Status.History[len(clusterVersion.Status.History)-1].CompletionTime
+		}
+	}
+	return nil
+}
+
+func getMatchedElementsFromMonitorEventMsg(regExp *regexp.Regexp, message string) (namespace, pod, node, reason, msg string) {
+	if regExp.MatchString(message) {
+		subMatches := regExp.FindStringSubmatch(message)
+		subNames := regExp.SubexpNames()
+		for i, name := range subNames {
+			switch name {
+			case "NS":
+				namespace = subMatches[i]
+			case "POD":
+				pod = subMatches[i]
+			case "NODE":
+				node = subMatches[i]
+			case "REASON":
+				reason = subMatches[i]
+			case "MSG":
+				msg = subMatches[i]
+			}
+		}
+	}
+	return
+}
+
+// isEventDuringInstallation returns true if the monitorEvent represents a real event that happened after installation.
+// regExp defines the pattern of the monitorEvent message. Named match is used in the pattern using `(?P<>)`. The names are placed inside <>. See example below
+// `ns/(?P<NS>openshift-ovn-kubernetes) pod/(?P<POD>ovnkube-node-[a-z0-9-]+) node/(?P<NODE>[a-z0-9.-]+) - reason/(?P<REASON>Unhealthy) (?P<MSG>Readiness probe failed:.*$`
+func isEventDuringInstallation(monitorEvent monitorapi.EventInterval, kubeClientConfig *rest.Config, regExp *regexp.Regexp) bool {
+	if kubeClientConfig == nil {
+		// default to OK
+		return true
+	}
+	installCompletionTime := getInstallCompletionTime(kubeClientConfig)
+	if installCompletionTime == nil {
+		// default to OK
+		return true
+	}
+
+	message := fmt.Sprintf("%s - %s", monitorEvent.Locator, monitorEvent.Message)
+	namespace, pod, _, reason, msg := getMatchedElementsFromMonitorEventMsg(regExp, message)
```

Comment:
if you don't get what you need, you need to explicitly fail or default ok or something.

Reply:
> if you don't get what you need, you need to explicitly fail or default ok or something.
Reading through, I recommend a recognizeable error. When this fails, it means a rather complicated regex failed.  So we want to fail early.

Reply:
Do you mean to fail the test or default to ok? For the case of allowedRepeatedEventFns, this is only used to return true/false.